src/groundlight/internalapi.py

```diff
@@ -70,6 +73,49 @@ class GroundlightApiClient(ApiClient):
     """
 
     REQUEST_ID_HEADER = "X-Request-Id"
+    MAX_CONNECTION_RETRY_ATTEMPTS = 4
+    # Initial delay time of 100 milliseconds
+    RETRY_INITIAL_DELAY = 0.1
+    RETRY_EXPONENTIAL_BACKOFF = 2
+    some_variable = 0
+
+    # Defines the range of HTTP error status codes for which the API
+    # will attempt to re-establish a connection
+    STATUS_CODE_RANGE = range(500, 600)
+
+    @staticmethod
+    def _retry_http_connection(callable):
```

Comment:
Let's pull the decorator out into its own class.  That makes it easier to understand and test.